### ISSUE TITLE
Keyframe scoping fails when declaring multiple values for the animation property.

### ISSUE THREAD

**Issue by HappyMajor** [angular/angular#49384]
### Which @angular/* package(s) are the source of the bug?

compiler

### Is this a regression?

No

### Description

I am using the default value of View.encapsulation and get the following problem:

![Screenshot 2023-03-03 103440](https://user-images.githubusercontent.com/1957550/224280638-7861136a-71bb-4bde-9386-45f91c84e8c0.png)

This will only scope the first animation theoFlyInX.

Chrome dev console:
![scope-failed](https://user-images.githubusercontent.com/1957550/224280651-1b65d406-f0bd-4f35-814c-a52d87371f23.png)

If I declare multiple animations like this then it will work:
![Screenshot 2023-03-03 103954](https://user-images.githubusercontent.com/1957550/224278119-fb693aae-1bf0-4c7e-80da-00a7da1d25cd.png)

Chrome dev console:
![scoped-right](https://user-images.githubusercontent.com/1957550/224278188-24aa7a11-29a0-4727-ac0b-f5c8e8b63755.png)

I think the problem lies in the following file: https://github.com/angular/angular/blob/a50ff60bff64859e4cd279a9ce0c8b47e6dfa691/packages/compiler/src/shadow_css.ts#L303

**Comment by dario-piotrowicz** [comment]
Hey @HappyMajor good luck with it :smiley: 

I did implement the keyframe scoping so I am familiar with how this works, please let me know if you need some help :slightly_smiling_face: 

(or if you'll want to hand it over, I'm happy to fix it as well :slightly_smiling_face:)

**Comment by HappyMajor** [comment]
@dario-piotrowicz Thanks for the response. I would like to try it, if I can not get it to work by tomorrow, I will hand it over to not delay the process, if thats okay? :)

**Comment by dario-piotrowicz** [comment]
@HappyMajor there is no ruch, feel free to tackle it (I'd also wouldn't have the time to properly dig into this in the next few days anyways), I was just offering to take it over just in case (in the past I did see some people taking up some issues but then dropping them for a reason or another so I just meant in that sense, if you want to bring this home it's all yours don't worry :muscle:)

If you need anything I'm here :slightly_smiling_face::+1:

**Comment by BrunoElo** [comment]
Experiencing this issue

**Comment by dario-piotrowicz** [comment]
@HappyMajor Did you manage to have a look into it? :slightly_smiling_face:

**Comment by vbraun** [comment]
Perhaps unsurprisingly, this is from css minification and you only have to turn on *that* optimization in angular.json` to encounter the bug
```
                        [...]
                        "optimization": {
                            "scripts": false,
                            "fonts": false,
                            "styles": {
                                "minify": true
                            }
                        },
```
Also, still bugged in Angular 17

**Comment by puckowski** [comment]
I think #56800 may have fixed this issue. In your screenshot, subsequent animation names were not separated by a space due to a bug which resulted in scoping issues prior to the aforementioned PR being merged.

On:
```
 "@angular/compiler": "^18.2.0-next.0",
```

I get:

```css
@keyframes _ngcontent-ng-c3888386579_theoFlyInX {
	0% {
		transform: translate(-100%);
		opacity: 0
	}

	to {
		transform: translate(0);
		opacity: 1
	}
}

@keyframes _ngcontent-ng-c3888386579_fadeInTheo {
	0% {
		opacity: 0
	}

	to {
		opacity: 1
	}
}

@keyframes _ngcontent-ng-c3888386579_theoFlyInY {
	0% {
		transform: translateY(-100%);
		opacity: 0
	}

	to {
		transform: translateY(0);
		opacity: 1
	}
}

@keyframes _ngcontent-ng-c3888386579_theoRotation {
	0% {
		transform: rotate(0)
	}

	to {
		transform: rotate(360deg)
	}
}

@keyframes _ngcontent-ng-c3888386579_theoGrow {
	0% {
		transform: scale(0);
		opacity: 0
	}

	to {
		transform: scale(1);
		opacity: 1
	}
}

#first[_ngcontent-ng-c3888386579] {
	animation: _ngcontent-ng-c3888386579_theoFlyInX .5s linear forwards, _ngcontent-ng-c3888386579_fadeInTheo .25s ease-in forwards, _ngcontent-ng-c3888386579_theoFlyInY .5s cubic-bezier(0, .5, 1, .9) forwards, _ngcontent-ng-c3888386579_theoRotation .2s ease-in-out 0s 3 alternate-reverse, _ngcontent-ng-c3888386579_theoGrow .3s ease-in-out .45s
}
```

In prod build:

```css
#first[_ngcontent-%COMP%] {
	animation: _ngcontent-%COMP%_theoFlyInX .5s linear forwards, _ngcontent-%COMP%_fadeInTheo .25s ease-in forwards, _ngcontent-%COMP%_theoFlyInY .5s cubic-bezier(0, .5, 1, .9) forwards, _ngcontent-%COMP%_theoRotation .2s ease-in-out 0s 3 alternate-reverse, _ngcontent-%COMP%_theoGrow .3s ease-in-out .45s
}
```

for:

```css
@keyframes theoFlyInX {
  from {
    transform: translateX(-100%);
    opacity: 0;
  }
  to {
    transform: translateX(0);
    opacity: 1;
  }
}

@keyframes fadeInTheo {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes theoFlyInY {
  from {
    transform: translateY(-100%);
    opacity: 0;
  }
  to {
    transform: translateY(0);
    opacity: 1;
  }
}

@keyframes theoRotation {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@keyframes theoGrow {
  from {
    transform: scale(0);
    opacity: 0;
  }
  to {
    transform: scale(1);
    opacity: 1;
  }
}

#first {
  animation: 
    theoFlyInX 0.5s linear forwards, 
    fadeInTheo 0.25s ease-in forwards, 
    theoFlyInY 0.5s cubic-bezier(0,0.5,1,0.9) forwards, 
    theoRotation 0.20s ease-in-out 0s 3 alternate-reverse,
    theoGrow 0.3s ease-in-out 0.45s;
}
```

**Comment by puckowski** [comment]
Hello @pkozlowski-opensource 

I was taking a look at existing issues. I believe this issue may be closed as resolved.

Can check out GitHub repo https://github.com/puckowski/angularissue49384 (uses compiler ```18.2.0-next.0```) or refer to my above comment.

I believe this was resolved by #56800
